### ISSUE TITLE
DB Handler needs to support multiple connections

### ISSUE THREAD

**Issue by jozefdobos** [3drepo/3drepogui#52]
Currently, the GUI will crash as soon as more than one DB action is attempted at the same time, eg:
- Connect and while loading all DBs show User Manager
- Connect and while loading all DBs load head of a 3D model
- etc

**Comment by carmenfan** [comment]
The crashing is fixed (requires the commit referenced within this issue, which is also merged into bouncer master branch). But if you dispatch simultaneous requests is very slow (because both workers require a lot of database access so they are constantly fighting with each other over that single connection to the database to get their data) but it shouldn't crash now and it should eventually load (it's faster to wait on one to complete before starting another...)

This should be relieved if you increase the amount of manipulator/database connection resources available for the gui, which is determined on the RepoController constructor:

````c++
		/**
		* Constructor
		* @param numConcurrentOps maximum number of requests it can handle concurrently
		* @param numDBConn number of concurrent connections to the database
		*/
		RepoController(
			std::vector<lib::RepoAbstractListener*> listeners = std::vector<lib::RepoAbstractListener *>(),
			const uint32_t &numConcurrentOps=1,
			const uint32_t &numDbConn=1);
````

....But it  doesn't at the moment... so I'll look into it (under this issue https://github.com/3drepo/3drepobouncer/issues/39)

**Comment by carmenfan** [comment]
Update: if you take bouncer after this commit: https://github.com/3drepo/3drepobouncer/commit/92f138d0af74ff405fb4f1f0e053e42545b95839

It should be a lot faster even with a single db conenction/manipulator. Having a 5s busy wait on a lock when a computer is running millions of clock cycles per second is a very stupid thing to do....
